lib/gush/client.rb

```diff
@@ -8,12 +8,18 @@ class Client
     @@redis_connection = Concurrent::ThreadLocalVar.new(nil)
 
     def self.redis_connection(config)
-      cached = (@@redis_connection.value ||= { url: config.redis_url, connection: nil })
-      return cached[:connection] if !cached[:connection].nil? && config.redis_url == cached[:url]
+      cached = (@@redis_connection.value ||= config.redis_opts.merge({ connection: nil }))
+      return cached[:connection] if !cached[:connection].nil? && config.redis_opts[:url] == cached[:url]
 
-      Redis.new(url: config.redis_url).tap do |instance|
+      # Support 4.2 and 4.1 https://github.com/redis/redis-rb/blame/v4.2.1/CHANGELOG.md
+      # These versions support ruby > 2.2
+      if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.3.0')
```

Comment:
Why was this based on Ruby version, and not redis-rb gem version? Feels like a mistake